projects/super_res/model/autoreg_diffusion_mod.py

```diff
@@ -0,0 +1,1442 @@
+import os
+import math
+from pathlib import Path
+from random import random
+from functools import partial
+from collections import namedtuple
+from joblib import Parallel, delayed
+
+import numpy as np
+
+import torch
+from torch import nn, einsum
+import torch.nn.functional as F
+import wandb
+
+import piq
+
+from kornia import filters
+from torch.optim import Adam
+
+from einops import rearrange, reduce
+from einops.layers.torch import Rearrange
+
+from PIL import Image
+
+import matplotlib as mpl
+from matplotlib.cm import ScalarMappable as smap
+
+from tqdm.auto import tqdm
+from ema_pytorch import EMA
+
+import flow_vis
+
+from accelerate import Accelerator
+
+from .network_swinir import SwinIR as net
+
+# constants
+
+ModelPrediction =  namedtuple('ModelPrediction', ['pred_noise', 'pred_x_start'])
+
+# helpers functions
+
+def calculate_crps(truth, pred, num_samples, num_videos_per_batch, num_frames, img_channels, img_size):
+    truth_cdf = np.zeros((256, 1, num_videos_per_batch, num_frames, img_channels, img_size, img_size), dtype = 'uint8')
+    for i in range(256):
```

Comment:
So all the images are stored with 8 bit precision, hence the 256. Leads to natural binning of the pixel space so calculating the cdf becomes very trivial in native numpy, otherwise there's an unnecessary layer of complexity